nix/cargo-doc-live.nix

```diff
@@ -0,0 +1,83 @@
+{ pkgs, lib, name, config, ... }:
+let
+  inherit (lib) types;
+in
+{
+  options = {
+    enable = lib.mkEnableOption name;
+
+    src = lib.mkOption {
+      type = types.path;
+      description = "The src of the cargo";
+    };
+
+    port = lib.mkOption {
+      type = types.port;
+      description = "The port for 'cargo doc'";
+      default = 8008;
+    };
+
+    crateName = lib.mkOption {
+      type = types.str;
+      description = "The crate to use when opening docs in browser";
+      default = builtins.replaceStrings [ "-" ] [ "_" ]
+        ((lib.trivial.importTOML "${config.src}/Cargo.toml").package.name);
+      defaultText = "The crate name is derived from the Cargo.toml file";
+    };
+
+    outputs.settings = lib.mkOption {
+      type = types.deferredModule;
+      internal = true;
+      readOnly = true;
+      default = {
+        processes =
+          {
+            "${name}-cargo-doc" = {
+              command = pkgs.writeShellApplication {
+                name = "cargo-doc";
+                runtimeInputs = with pkgs; [ cargo cargo-watch nodePackages.browser-sync ];
+                text =
+                  let
+                    browser-sync = lib.getExe pkgs.nodePackages.browser-sync;
+                    cargo-watch = lib.getExe pkgs.cargo-watch;
+                    cargo = lib.getExe pkgs.cargo;
+                  in
```

Comment:
Since these are in `runtimeInputs` already, we don't need these and can just reference the executables as is in the script text, no?